x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/optimizer/OptimizerTests.java

```diff
@@ -698,26 +687,24 @@ public void testSimplifyCaseConditionsFoldWhenTrue() {
         // ELSE 'default'
         // END
 
+        SimplifyCase rule = new SimplifyCase();
         Case c = new Case(EMPTY, Arrays.asList(
             new IfConditional(EMPTY, new Equals(EMPTY, getFieldAttribute(), ONE), Literal.of(EMPTY, "foo1")),
             new IfConditional(EMPTY, new Equals(EMPTY, ONE, ONE), Literal.of(EMPTY, "bar1")),
             new IfConditional(EMPTY, new Equals(EMPTY, TWO, ONE), Literal.of(EMPTY, "bar2")),
             new IfConditional(EMPTY, new GreaterThan(EMPTY, getFieldAttribute(), ONE), Literal.of(EMPTY, "foo2")),
             Literal.of(EMPTY, "default")));
-        assertFalse(c.foldable());
-
-        SimplifyCase rule = new SimplifyCase();
         Expression e = rule.rule(c);
         assertEquals(Case.class, e.getClass());
         c = (Case) e;
         assertEquals(2, c.conditions().size());
-        assertThat(c.conditions().get(0).condition().toString(), startsWith("Equals[a{f}#"));
-        assertThat(c.conditions().get(1).condition().toString(), startsWith("Equals[=1,=1]#"));
+        assertThat(c.conditions().get(0).condition().getClass(), is(Equals.class));
+        assertThat(c.conditions().get(1).condition().getClass(), is(Equals.class));
         assertFalse(c.foldable());
         assertEquals(TypeResolution.TYPE_RESOLVED, c.typeResolved());
     }
 
-    public void testSimplifyCaseConditionsFoldCompletely_FoldableElse() {
+    public void testSimplifyCaseConditionsFoldCompletely() {
```

Comment:
Why did you make these changes?
They were added because of a bug when the conditions were folded but the results could not be folded.